File: converted/classes_only/classes_only/StateMachine$SubstateSlot.java
package zombie.ai;

public class StateMachine$SubstateSlot {
   private State state;
   boolean shouldBeActive;

   StateMachine$SubstateSlot(State var1) {
      this.state = var1;
      this.shouldBeActive = true;
   }

   public State getState() {
      return this.state;
   }

   void setState(State var1) {
      this.state = var1;
   }

   public boolean isEmpty() {
      return this.state == null;
   }
}

File: ProjectZomboidSwitch/src/zombie/ai/StateMachine$SubstateSlot.cpp
#include "zombie/ai/StateMachine$SubstateSlot.h"

namespace zombie {
namespace ai {

State StateMachine$SubstateSlot::getState() {
    return state;
}

void StateMachine$SubstateSlot::setState(State var1) {
    state = var1;
}

bool StateMachine$SubstateSlot::isEmpty() {
    return state == nullptr;
}

} // namespace ai
} // namespace zombie
